src/main/java/com/google/sps/DataServlet.java

```diff
@@ -169,16 +161,38 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     }
     truncatedGraph = currDataGraph.getReachableNodes(nodeNameParam, depthNumber);
 
-    // We set the headers in the following 3 scenarios:
+    if (nodeNameParam.equals("") && truncatedGraph.equals(currDataGraph.graph())) {
+      // If we are not filtering the graph or limiting its depth, show all mutations of all nodes
+      filteredMutationIndices = defaultIndices;
+    } else {
+      // Get the names of all the displayed nodes and find all indices of mutations
+      // that mutate any of them
+      Set<String> truncatedGraphNodeNames = Utility.getNodeNamesInGraph(truncatedGraph);
+      // Also get mutations relevant to the searched node if it is not an empty string
+      if (nodeNameParam.length() != 0) {
+        truncatedGraphNodeNames.add(nodeNameParam);
```

Comment:
if nodeNameParam is not empty (something is searched),  shouldn't it already be in the truncated graph? (since nodeNameParam was passed into the collection that the reachable nodes function took)

Reply:
I think this was just for highlighting the diff in case the searched node was deleted. Open to better ways of doing this :)

Reply:
I think this is what we worked off of for other branches so it seems fine!!